test_suite/gap/test_start_stop_scan.py

```diff
@@ -103,3 +104,31 @@ def test_scan_should_not_report_advertisement_when_peer_inactive(scanner: BleDev
     scanner.gap.setScanParameters()
     scans = scanner.gap.scanForAddress(advertiser_address, scan_timeout).result
     assert len(scans) == 0
+
+
+@pytest.mark.ble41
+def test_scan_start_stop_can_be_called_in_succession(scanner: BleDevice):
+    """startScan should not report scan results when there is no device advertising"""
+    # start scan
+
+    scanner.gap.startScan(0, 'DISABLE', 0)
+    assert scanner.gap.isRadioActive().result is True
+
+    scanner.gap.stopScan()
+    # this is async so we wait
+    sleep(0.5)
+    assert scanner.gap.isRadioActive().result is False
+
+    for i in range(10):
+        scanner.gap.stopScan()
+        scanner.gap.startScan(0, 'DISABLE', 0)
+
+    assert scanner.gap.isRadioActive().result is True
```

Comment:
Is there enough time to start the scan between this command and the last one ?

Reply:
Yes, starting is immediate as it immediately goes to pending which is enough as scan is not idle anymore.